lib/webhooks/__tests__/register.test.ts

```diff
@@ -291,13 +295,70 @@ describe('shopify.webhooks.register', () => {
         JSON.stringify(mockResponses.webhookCheckEmptyResponse),
       );
 
-      const response = await shopify.webhooks.register(REGISTER_PARAMS);
+      const response = await shopify.webhooks.register({session});
 
       expect(mockTestRequests.requestList).toHaveLength(1);
       expect(response[gdprTopic]).toHaveLength(0);
       expect(shopify.webhooks.getTopicsAdded()).toContain(gdprTopic);
     });
   });
+
+  it('deletes handlers not currently in the registry', async () => {
+    const topic = 'NEW_TOPIC_TO_ADD';
+    const handler = {...HTTP_HANDLER};
+    const responses = [
+      mockResponses.successResponse,
+      mockResponses.successDeleteResponse,
+    ];
+
+    const registerReturn = await registerWebhook({
+      topic,
+      handler,
+      checkMockResponse: mockResponses.webhookCheckResponse,
+      responses,
+    });
+
+    expect(Object.keys(registerReturn)).toEqual([
+      'NEW_TOPIC_TO_ADD',
+      'PRODUCTS_CREATE',
+    ]);
+  });
+
+  it('returns multiple results per topic for multiple webhooks per topic', async () => {
+    const topic = 'PRODUCTS_CREATE';
+    const httpHandler = HTTP_HANDLER;
+    const ebHandler = EVENT_BRIDGE_HANDLER;
+    const responses = [
+      mockResponses.successResponse,
+      mockResponses.eventBridgeSuccessResponse,
+    ];
+
+    shopify.webhooks.addHandlers({[topic]: httpHandler});
+    shopify.webhooks.addHandlers({[topic]: ebHandler});
+
+    queueMockResponse(JSON.stringify(mockResponses.webhookCheckEmptyResponse));
+    responses.forEach((response) => {
+      queueMockResponse(JSON.stringify(response));
+    });
+    const registerReturn = await shopify.webhooks.register({session});
+    console.log(
+      `DEBUG: registerReturn: ${JSON.stringify(registerReturn, null, 2)}`,
+    );
+    expect(mockTestRequests.requestList).toHaveLength(responses.length + 1);
+    assertWebhookCheckRequest({session});
+
+    assertWebhookRegistrationRequest(
+      'webhookSubscriptionCreate',
+      `topic: ${topic}`,
+      {callbackUrl: `"https://test_host_name/webhooks"`},
+    );
+    assertWebhookRegistrationRequest(
+      'eventBridgeWebhookSubscriptionCreate',
+      `topic: ${topic}`,
+      {arn: '"arn:test"'},
+    );
+    assertRegisterResponse({registerReturn, topic, responses});
+  });
```

Comment:
I was trying to wrap my head around the return result content from register when there are multiple handlers registered per topic (e.g., HTTP and Event Bridge), so I added in this test.

There's probably more derivatives of this test that should be added, e.g., for a topic with multiple handlers
- what happens when the HTTP handler changes but the EB stays the same
- what happens when one of the handlers for a given topic is deleted
- etc.

Reply:
No objections here!